src/Stratis.Bitcoin.Features.Consensus/CoinViews/Coindb/LeveldbCoindb.cs

```diff
@@ -295,41 +300,47 @@ public int GetMinRewindHeight()
         /// <inheritdoc />
         public HashHeightPair Rewind(HashHeightPair target)
         {
-            using (var batch = new WriteBatch())
-            {
-                HashHeightPair current = this.GetTipHash();
-                return RewindInternal(batch, current.Height);
-            }
+            HashHeightPair current = this.GetTipHash();
+            return RewindInternal(current.Height, target);
         }
 
-        private HashHeightPair RewindInternal(WriteBatch batch, int height)
+        private HashHeightPair RewindInternal(int startHeight, HashHeightPair target)
         {
-            byte[] row = this.leveldb.Get(new byte[] { rewindTable }.Concat(BitConverter.GetBytes(height).Reverse()).ToArray());
+            HashHeightPair res = null;
 
-            if (row == null)
-                throw new InvalidOperationException($"No rewind data found for block at height {height}.");
+            using (var batch = new WriteBatch())
+            {
+                for (int height = startHeight; height > (target?.Height ?? (startHeight - 1)) && height > (startHeight - MaxRewindBatchSize); height--)
```

Comment:
It might be wrong but if you pass in a null for target then this for wont execute as height will be greater than startheight?

Maybe consider determining end height for the `for` loop in a variable outside of the for loop so that is a bit more readible ;)

Reply:
@fassadlr . the loop is decreasing. The idea is for the loop to execute exactly once to match legacy behaviour when `null` is passed from the test cases.